### PR TITLE
Feature/10349 crypto functions part1/10575 extend http client dynamic pinning

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
added rest client for dynamic pinning.
refactoring to extract evaluate trust method and init URLSessionDelegate with them
added tests to check if an update of EvaluateTrust works

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10575

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/HTTPClient/Security/EvaluateTrust/DefaultEvaluateTrust.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			// Matching fingerprint?
+			let keyHash = serverPublicKeyData.sha256String()
+			if publicKeyHash == keyHash {
+				// Success! This is our server
+				completionHandler(.useCredential, URLCredential(trust: trust))
+				return
+			}
+		}
+		
+		completionHandler(.cancelAuthenticationChallenge, /* credential */ nil)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/HTTPClient/Security/EvaluateTrust/DefaultEvaluateTrust.swift:45]: For succes and error cases i am missing some logs.

File: src/xcode/ENA/ENA/Source/HTTPClient/Security/EvaluateTrust/DefaultEvaluateTrust.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		
+		// we expect a chain of at least 2 certificates
+		// index '1' is the required intermediate
+		if let serverCertificate = SecTrustGetCertificateAtIndex(trust, 1),
+		   let serverPublicKey = SecCertificateCopyKey(serverCertificate),
+		   let serverPublicKeyData = SecKeyCopyExternalRepresentation(serverPublicKey, nil ) as Data? {
+			
+			// Matching fingerprint?
+			let keyHash = serverPublicKeyData.sha256String()
+			if publicKeyHash == keyHash {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/HTTPClient/Security/EvaluateTrust/DefaultEvaluateTrust.swift:38]: when the condition fails, don't we need to call the completion handler, too, but with an error?

File: src/xcode/ENA/ENA/Source/HTTPClient/Service/RestServiceProvider.swift [hunk 3]
```diff
@@ -26,9 +31,12 @@ class RestServiceProvider: RestServiceProviding {
 		environment: EnvironmentProviding = Environments(),
 		session: URLSession? = nil
 	) {
+		self.environment = environment
+		self.optionalSession = session
 		self.restService = StandardRestService(environment: environment, session: session)
 		self.cachedService = CachedRestService(environment: environment, session: session, cache: KeyValueCacheFake())
 		self.wifiOnlyService = WifiOnlyRestService(environment: environment, session: session)
+		self.dynamicPinningRestService = DynamicPinningRestService(environment: environment, session: session)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/HTTPClient/Service/RestServiceProvider.swift:39]:
```suggestion
		self.dynamicPinningService = DynamicPinningService(environment: environment, session: session)
```

Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/HTTPClient/Service/RestServiceProvider.swift:39]: `DynamicPinningService` might be misleading, it's a new RestService. I think we should keep the name.

**Review by 30mar — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**